.pipelines/e2e-job-template.yaml

```diff
@@ -0,0 +1,37 @@
+parameters:
+  name: ''  # defaults for any parameters that aren't specified
+  k8sRelease: ''
+  apimodel: 'examples/e2e-tests/kubernetes/release/default/definition.json'
+  createVNET: false
+
+jobs:
+- job: ${{ parameters.name }}
+  dependsOn: unit_tests
+  timeoutInMinutes: 90
+  cancelTimeoutInMinutes: 5
+  strategy:
+    maxParallel: 0
+  pool:
+    vmImage: ubuntu-16.04
+  
+  container: dev1
+  
+  variables:
+    GOBIN:  '$(GOPATH)/bin' # Go binaries path
+    GOROOT: '/usr/local/go' # Go installation path
+    GOPATH: '$(system.defaultWorkingDirectory)/gopath' # Go workspace path
+    modulePath: '$(GOPATH)/src/github.com/$(build.repository.name)' # Path to the module's code
+    TIMEOUT: 20m
+    CLUSTER_DEFINITION: ${{ parameters.apimodel }}
+    ORCHESTRATOR_RELEASE: ${{ parameters.k8sRelease }}
+    CREATE_VNET: ${{ parameters.createVNET }}
+    CLEANUP_ON_EXIT: true
+    CLEANUP_IF_FAIL: true
+    RETAIN_SSH: false
+    ENABLE_KMS_ENCRYPTION: true
+    SUBSCRIPTION_ID: '$(SUBSCRIPTION_ID_E2E_KUBERNETES)'
```

Comment:
Lets rename this to AZURE_SUBSCRIPTION_ID. This seems to be the env naming convention

Reply:
This is defined in by our e2e tests https://github.com/Azure/aks-engine/blob/master/test/e2e/azure/cli.go#L35 and is the same as circle uses

Reply:
Then let's do away this `SUBSCRIPTION_ID_E2E_KUBERNETES`. I assume that variable is being declared in vsts internally. Let's do that internal declaration for `SUBSCRIPTION_ID` itself.

Reply:
Sure. I just copied that over from circle. I will still need to declare the var here for it to be available in the container job. I assume the idea behind e2e_kubernetes was that we have different variables subscriptions declared in circle env vars. For now this is the only one in vsts though so I'm okay with changing it.

Reply:
Actually makes sense. We can have this PR as is for now